### PR TITLE
azure scale set infrastructure documentation

### PR DESCRIPTION
**Pull request by luispinedamx**
- Add Azure Scale Set infrastructure documentation
- some minor corrections

### CODE REVIEW

File: src/docs/admin/references/NodeSourceInfrastructureReference.adoc [hunk 1]
```diff
@@ -588,7 +588,96 @@ The next four parameters are optional, and apply to advanced Azure endpoint conf
...
+- *region:* Define the Azure region where the Resource Group and the Scale Set will be created. For a list of the regions supported by your subscription, you can use the https://docs.microsoft.com/en-us/cli/azure/account?view=azure-cli-latest#az-account-list-locations[CLI command] `az account list-locations -o table`, the right name to use is the one in the column _name_. The region name should be in lowercase without spaces, e.g. for the region _West US 2_ use _westus2_. Default value is _westeurope_.
+- *rmUrl:* Optional parameter. Provide your server's URL or public IP (e.g. try.activeeon.com). If left unset, the default Resource Manager URL will be fetched from the system properties.
+- *rmCredentials:* Optional parameter. Provide credentials to connect to the Resource Manager. If left blank, the credentials of the logged user will be used.
+- *deploymentTimeout:* Delay before the Resource Manager can declare a deploying node as lost. To be set in _minutes_. You can set it to a reasonable time after which you expect the node to be running or marked as lost otherwise (e.g. _10_ minutes). Default value is _120_.
```
Comment (mcompastie) [inline src/docs/admin/references/NodeSourceInfrastructureReference.adoc]: Specifying the unit of the parameter can facilitate its understanding from the user.

File: src/docs/admin/references/NodeSourceInfrastructureReference.adoc [hunk 2]
```diff
@@ -588,7 +588,96 @@ The next four parameters are optional, and apply to advanced Azure endpoint conf
...
+- *cleaningDelay:* Periodical cleaning delay, expressed in _seconds_. Every _X_ seconds the Resource Manager will actually remove decommissioned instances from the Scale Set. Default value is _60_.
```
Comment (mcompastie) [inline src/docs/admin/references/NodeSourceInfrastructureReference.adoc]: Bis repetita

File: src/docs/admin/references/NodeSourceInfrastructureReference.adoc [hunk 3]
```diff
@@ -588,7 +588,96 @@ The next four parameters are optional, and apply to advanced Azure endpoint conf
...
+- *machineType:* Define the type of Azure VM that you want to use for your Scale Set. You can use any size from Azure's Standard Tier, provided you have enough quota and budget to launch them. The default value is _Standard_D1_v2_, you should provide a valid type using a similar syntax (case insensitive). Azure provides an extensive list of VM sizes for https://docs.microsoft.com/en-us/azure/virtual-machines/linux/sizes[Linux] and https://docs.microsoft.com/en-us/azure/virtual-machines/windows/sizes[Windows] VMs. If you want to know the available sizes for a specific region or subscription you can use https://docs.microsoft.com/en-us/cli/azure/vm?view=azure-cli-latest#az-vm-list-sizes[Azure CLI command] `az vm list-sizes -l {your location}`.
+- *osType:* A choice between _linux_ and _windows_. Default value is _linux_.
+- *Image:* You can either provide a custom Azure image, or select one of the available standard ones. To provide a custom image use the format _{resourceGroupName}:{imageName}_. If you prefer to use a generic image, you can choose from the following lists (provided by Azure API):
```
Comment (mcompastie) [inline src/docs/admin/references/NodeSourceInfrastructureReference.adoc]: A pointer to the Azure documentation could be a great complement for this section.

File: src/docs/admin/references/NodeSourceInfrastructureReference.adoc [hunk 4]
```diff
@@ -588,7 +588,96 @@ The next four parameters are optional, and apply to advanced Azure endpoint conf
...
+
+===== Pre-Requisites
+
+As in our standard <<_azure_infrastructure>>, ProActive relies on an existing Azure user account with a valid subscription in order to deploy a Scale Set. These are the specific requirements:
+
+- *Azure account:* You should have a valid Azure account, linked to an Azure subscription with enough budget and https://docs.microsoft.com/en-us/azure/azure-subscription-service-limits[service quotas] to deploy the Scale Set that you request.
+- *Azure privileges:* If your account does not have full Admin privileges on your subscription, please make sure you have rights to:
+* Create a https://docs.microsoft.com/en-us/azure/azure-resource-manager/resource-group-overview#resource-groups[resource group] in a region of your choice.
+* Create a Scale Set in that resource group, and create its underlying resources: Public IP Address, Load Balancer, Network Security Group and Virtual Network.
```
Comment (mcompastie) [inline src/docs/admin/references/NodeSourceInfrastructureReference.adoc]: Please ensure that every network prerequisite is formulated, e.g. outgoing connectivity to ProActive Resource Manager.

File: src/docs/admin/references/NodeSourceInfrastructureReference.adoc [hunk 5]
```diff
@@ -588,7 +588,96 @@ The next four parameters are optional, and apply to advanced Azure endpoint conf
...
+
+==== Azure Scale Set Infrastructure
+
+We provide a highly customizable infrastructure to deploy ProActive nodes using an https://docs.microsoft.com/en-us/azure/virtual-machine-scale-sets/overview[Azure Scale Set]. This infrastructure will allow you to automatically adjust the number of deployed virtual machines (VM) in response to the current workload, and according to your configured parameters, e.g. min/max number of nodes, number of nodes per VM, or minimal up-time. Auto-scaling will help you increase your processing capacity during peak computation periods, while keeping your expenses low during less intensive or idle periods.
+
+Whether you require to run your workflows on Linux or on Windows, use standard or custom VM images, do some pre-configuration steps, or mount a shared file system, the Azure Scale Set infrastructure can handle it. This guide describes the requirements and the various configuration parameters available to deploy an Azure Scale Set infrastructure.
```
Comment (MaelAudren) [inline src/docs/admin/references/NodeSourceInfrastructureReference.adoc]: I would remove the `on` for `on Windows` and the comma before `or`.
=> `Whether you require to run your workflows on Linux or Windows, use standard or custom VM images, do some pre-configuration steps or mount a shared file system, the Azure Scale Set infrastructure can handle it.`

File: src/docs/admin/references/NodeSourceInfrastructureReference.adoc [hunk 6]
```diff
@@ -588,7 +588,96 @@ The next four parameters are optional, and apply to advanced Azure endpoint conf
...
+* *_Windows:_* The commands will be included in a _.bat_ batch file and must therefore be DOS-compatible. As a guideline, here is an unofficial https://www.robvanderwoude.com/batchcommands.php[list of DOS commands] and a http://steve-jansen.github.io/guides/windows-batch-scripting/part-1-getting-started.html[scripting tutorial].
+- *jvmParameters:* Add JVM parameters to configure your ProActive node, use the format _-Dproperty=value_. The default values allow for a correct deployment of the node. You can add your own parameters but be aware that removing the current values might cause the deployment to fail. A non-exhaustive list of node-related JVM parameters can be found in your configuration file `$PROACTIVE_HOME/config/network/node.ini`
+- *armTemplateUrl:* Optional parameter, for advanced configuration only. Provide a URL with a valid _.json_ Azure Resource Management https://docs.microsoft.com/en-us/azure/azure-resource-manager/resource-group-authoring-templates[configuration file]. If set, this file will override the default ARM file.
```
Comment (MaelAudren) [inline src/docs/admin/references/NodeSourceInfrastructureReference.adoc]: The ARM file is a windows file that enable to set the group authorization ? If it is not the case please add explanation about the ARM file.

File: src/docs/admin/references/NodeSourceInfrastructureReference.adoc [hunk 7]
```diff
@@ -588,7 +588,96 @@ The next four parameters are optional, and apply to advanced Azure endpoint conf
...
+
+As you can see, these parameters provide a lot of flexibility to configure your infrastructure. When creating your Azure Scale Set node source, the infrastructure should be coupled with a Dynamic Policy. This Policy will additionally define scalability parameters such as limits on the number of deployed nodes, or the minimum idle time before a node can be deleted (to optimize node utilization).
```
Comment (MaelAudren) [inline src/docs/admin/references/NodeSourceInfrastructureReference.adoc]: I would remove the comma before or
=> `This Policy will additionally define scalability parameters such as limits on the number of deployed nodes or the minimum idle time before a node can be deleted (to optimize node utilization).`

File: src/docs/admin/references/NodeSourceInfrastructureReference.adoc [hunk 8]
```diff
@@ -764,7 +853,7 @@ The infrastructure parameters are described hereafter:
  ** `sudo`: similar to `none` regarding the connection to the head node, but a `sudo` command will be initiated to impersonate as the current <<_glossary_proactive_scheduler_user,ProActive Scheduler user>>, before doing a job submission.
  * *alternateRMUrl* : the url used by the ProActive Nodes to contact <<_glossary_resource_manager,ProActive Resource Manager>>. This url is displayed on ProActive server startup. Example: `pnp://myserver:64738`.
```
Comment (MaelAudren) [inline src/docs/admin/references/NodeSourceInfrastructureReference.adoc]: `url`=> `URL`

File: src/docs/admin/references/NodeSourceInfrastructureReference.adoc [hunk 9]
```diff
@@ -764,7 +853,7 @@ The infrastructure parameters are described hereafter:
...
  * *sshPort* : port used for ssh connections.
```
Comment (MaelAudren) [inline src/docs/admin/references/NodeSourceInfrastructureReference.adoc]: `ssh` => `SSH`

Comment (luispinedamx) [reply src/docs/admin/references/NodeSourceInfrastructureReference.adoc]: Agree, there is also another lowercase "url". Let me piggyback all these corrections.

File: src/docs/admin/references/NodeSourceInfrastructureReference.adoc (hunk 8 above)
Comment (luispinedamx) [reply src/docs/admin/references/NodeSourceInfrastructureReference.adoc]: OK, although it's not part of my PR ;)

File: src/docs/admin/references/NodeSourceInfrastructureReference.adoc (hunk 6 above)
Comment (luispinedamx) [reply src/docs/admin/references/NodeSourceInfrastructureReference.adoc]: > The ARM file is a windows file that enable to set the group authorization ?
I don't understand from where did you infer this. The ARM templates are a whole topic, that's why I put a link to the documentation

File: src/docs/admin/references/NodeSourceInfrastructureReference.adoc (hunk 5 above)
Comment (luispinedamx) [reply src/docs/admin/references/NodeSourceInfrastructureReference.adoc]: - Ok for the "on".
- For the comma I prefer to keep it as it really helps breaking correctly the elements on the list. I understand it's a matter of taste, it's an Oxford comma dilemma. ;)
https://english.stackexchange.com/questions/267256/oxford-comma-before-or

File: src/docs/admin/references/NodeSourceInfrastructureReference.adoc (hunk 4 above)
Comment (luispinedamx) [reply src/docs/admin/references/NodeSourceInfrastructureReference.adoc]: In this case it's actually not strictly necessary: the boot script and more precisely the configuration file opens the necessary ports on the scale set VMs to connect to ProActive's RM.

File: src/docs/admin/references/NodeSourceInfrastructureReference.adoc (hunk 3 above)
Comment (luispinedamx) [reply src/docs/admin/references/NodeSourceInfrastructureReference.adoc]: Done, I added the links on the table headers.

File: src/docs/admin/references/NodeSourceInfrastructureReference.adoc (hunk 1 above)
Comment (luispinedamx) [reply src/docs/admin/references/NodeSourceInfrastructureReference.adoc]: It is mentioned "To be set in _minutes_".

File: src/docs/admin/references/NodeSourceInfrastructureReference.adoc (hunk 2 above)
Comment (luispinedamx) [reply src/docs/admin/references/NodeSourceInfrastructureReference.adoc]: You mean about "remove decommissioned"? I agree it's not clear (but it is actually what I meant). Let me rephrase it.

File: src/docs/admin/references/NodeSourceInfrastructureReference.adoc (hunk 1 above)
Comment (mcompastie) [reply src/docs/admin/references/NodeSourceInfrastructureReference.adoc]: Ok, I've missed the point.

File: src/docs/admin/references/NodeSourceInfrastructureReference.adoc (hunk 2 above)
Comment (mcompastie) [reply src/docs/admin/references/NodeSourceInfrastructureReference.adoc]: My point was not about about `decommissioned` but about the unit as well. Well, it's my mistake too here.

File: src/docs/admin/references/NodeSourceInfrastructureReference.adoc (hunk 4 above)
Comment (mcompastie) [reply src/docs/admin/references/NodeSourceInfrastructureReference.adoc]: Ok, maybe it would be nice to get a big picture of it.

File: src/docs/admin/references/NodeSourceInfrastructureReference.adoc (hunk 6 above)
Comment (luispinedamx) [reply src/docs/admin/references/NodeSourceInfrastructureReference.adoc]: Fixed it, let me know if it's clearer @MaelAudren
